components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```
@@ -200,6 +200,34 @@ public IdentityProviderListResponse getIDPs(String requiredAttributes, Integer l
         }
     }
 
+    /**
+     * Get list of identity providers.
+     *
+     * @param requiredAttributes Required attributes in the IDP list response.
+     * @param limit      Items per page.
+     * @param offset     Offset.
+     * @param filter     Filter string. E.g. filter="name" sw "google" and "isEnabled" eq "true"
+     * @param sortBy     Attribute to sort the IDPs by. E.g. name
+     * @param sortOrder  Order in which IDPs should be sorted. Can be either ASC or DESC.
+     * @return IdentityProviderListResponse.
+     */
+    public IdentityProviderListResponse getTrustedTokenIssuers(String requiredAttributes, Integer limit, Integer offset,
+                                                      String filter, String sortBy, String sortOrder) {
+
+        try {
+            List<String> requestedAttributeList = null;
+            if (StringUtils.isNotBlank(requiredAttributes)) {
+                requestedAttributeList = new ArrayList<>(Arrays.asList(requiredAttributes.split(",")));
+            }
+            return createIDPListResponse(
+                    IdentityProviderServiceHolder.getIdentityProviderManager().getTrustedTokenIssuers(limit, offset,
+                            filter, sortBy, sortOrder, ContextLoader.getTenantDomainFromContext(),
+                            requestedAttributeList), requestedAttributeList);
+        } catch (IdentityProviderManagementException e) {
+            throw handleIdPException(e, Constants.ErrorMessage.ERROR_CODE_ERROR_LISTING_IDPS, null);
```

Should we throw a different error?

---

Something to indicate this was about trusted token issuers?